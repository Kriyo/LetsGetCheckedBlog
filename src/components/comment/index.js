import React from 'react'

import {
  Container,
  Item,
  RootItem,
  NestedItem,
  Count,
  Image,
  DetailsWrapper,
  Content,
  AuthorDetails,
  ReplyHeading,
  Form,
  Button,
  ButtonWrapper,
  ErrorLabel,
  Input,
  TextArea,
  Section,
} from './styles/comment'

export const Comment = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Comment.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Comment.Section = ({ children, ...props }) => {
  return <Section {...props}>{children}</Section>
}

Comment.RootComment = ({ children, ...props }) => {
  return <RootItem {...props}>{children}</RootItem>
}

Comment.NestedComment = ({ children, ...props }) => {
  return <NestedItem {...props}>{children}</NestedItem>
}

Comment.Count = ({ children, ...props }) => {
  return <Count {...props}>{children}</Count>
}

Comment.DetailsWrapper = ({ children, ...props }) => {
  return <DetailsWrapper {...props}>{children}</DetailsWrapper>
}

Comment.AuthorDetails = ({ children, ...props }) => {
  return <AuthorDetails {...props}>{children}</AuthorDetails>
}

Comment.Avatar = ({ children, ...props }) => {
  return <Image {...props}>{children}</Image>
}

Comment.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}

Comment.ReplyHeading = ({ children, ...props }) => {
  return <ReplyHeading {...props}>{children}</ReplyHeading>
}

Comment.Form = ({ children, ...props }) => {
  return <Form {...props}>{children}</Form>
}

Comment.ButtonWrapper = ({ children, ...props }) => {
  return <ButtonWrapper {...props}>{children}</ButtonWrapper>
}

Comment.Button = ({ children, ...props }) => {
  return <Button {...props}>{children}</Button>
}
Comment.ErrorLabel = ({ children, ...props }) => {
  return <ErrorLabel {...props}>{children}</ErrorLabel>
}
Comment.UserInput = ({ children, ...props }) => {
  return <Input {...props}>{children}</Input>
}

Comment.TextArea = ({ children, ...props }) => {
  return <TextArea {...props}>{children}</TextArea>
}
