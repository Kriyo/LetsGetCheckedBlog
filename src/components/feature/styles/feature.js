import styled from 'styled-components/macro'
import { Primary } from '../../../utils/colors'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  border-bottom: 8px solid ${Primary[300]};
  text-align: center;
  padding: 355px 45px;
  @media (max-width: 600px) {
    padding: 165px 45px;
  }
`

export const Title = styled.h1`
  color: white;
  max-width: 640px;
  font-size: 3rem;
  font-weight: 500;
  margin: auto;
  @media (max-width: 600px) {
    font-size: 2rem;
  }
`

export const SubTitle = styled.h2`
  color: white;
  font-size: 1.6rem;
  font-weight: normal;
  margin: 16px auto;
  @media (max-width: 600px) {
    font-size: 1.3rem;
  }
`
