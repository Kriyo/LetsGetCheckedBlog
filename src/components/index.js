export { Comment } from './comment'
export { Feature } from './feature'
export { NavBar } from './navbar'
export { Post } from './post'
