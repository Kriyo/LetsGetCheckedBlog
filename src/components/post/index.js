import React from 'react'

import {
  Container,
  Item,
  Author,
  Title,
  Description,
  PublishDate,
  Content,
} from './styles/post'

export const Post = ({ children, ...props }) => {
  return <Item {...props}>{children}</Item>
}

Post.Frame = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Post.Author = ({ children, ...props }) => {
  return <Author {...props}>{children}</Author>
}

Post.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Post.Description = ({ children, ...props }) => {
  return <Description {...props}>{children}</Description>
}

Post.PublishDate = ({ children, ...props }) => {
  return <PublishDate {...props}>{children}</PublishDate>
}

Post.Content = ({ children, ...props }) => {
  return <Content {...props}>{children}</Content>
}
