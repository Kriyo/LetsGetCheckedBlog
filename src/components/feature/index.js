import React from 'react'

import { Container, Title, SubTitle } from './styles/feature'

export const Feature = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

Feature.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Feature.SubTitle = ({ children, ...props }) => {
  return <SubTitle {...props}>{children}</SubTitle>
}
