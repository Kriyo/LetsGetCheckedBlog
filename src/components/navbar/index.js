import React from 'react'
import { Background, Container, Link, Logo } from './styles/navbar'

export default function NavBar({ background = true, children, ...props }) {
  return background ? <Background {...props}>{children}</Background> : children
}

NavBar.Frame = function NavBarFrame({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

NavBar.Link = function NavBarLink({ children, ...props }) {
  return <Link {...props}>{children}</Link>
}

NavBar.Logo = function NavBarLogo({ to, ...restProps }) {
  return <Logo {...restProps} />
}
