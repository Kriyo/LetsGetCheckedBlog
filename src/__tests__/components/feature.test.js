import React from 'react'
import { render } from '@testing-library/react'
import { Feature } from '../../components/feature'

describe('<Feature />', () => {
  it('renders the <Feature /> with populated data', () => {
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>Simple Blog</Feature.Title>
        <Feature.SubTitle>
          At vero eos et accusamus et iusto odio
        </Feature.SubTitle>
      </Feature>
    )

    expect(getByText('Simple Blog')).toBeTruthy()
    expect(getByText('At vero eos et accusamus et iusto odio')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
