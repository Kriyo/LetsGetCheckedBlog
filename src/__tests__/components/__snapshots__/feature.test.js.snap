// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-uyg3os-0 kriDBt"
>
  <h1
    class="feature__Title-uyg3os-1 gDAVhI"
  >
    Simple Blog
  </h1>
  <h2
    class="feature__SubTitle-uyg3os-2 juyoCa"
  >
    At vero eos et accusamus et iusto odio
  </h2>
</div>
`;
